cmd/smtp-notify/main.go

```diff
@@ -0,0 +1,320 @@
+	defDBUser        = "mainflux"
+	defDBPass        = "mainflux"
+	defDB            = "subscriptions"
+	defConfigPath    = "/config.toml"
+	defDBSSLMode     = "disable"
+	defDBSSLCert     = ""
+	defDBSSLKey      = ""
+	defDBSSLRootCert = ""
+	defHTTPPort      = "8180"
+	defServerCert    = ""
+	defServerKey     = ""
+	defJaegerURL     = ""
+	defNatsURL       = "nats://localhost:4222"
+
+	defEmailDriver      = "smtp"
+	defEmailHost        = "localhost"
+	defEmailPort        = "25"
+	defEmailUsername    = "root"
+	defEmailPassword    = ""
+	defEmailFromAddress = ""
+	defEmailFromName    = ""
+	defEmailTemplate    = "email.tmpl"
+
+	defAuthTLS     = "false"
+	defAuthCACerts = ""
+	defAuthURL     = "localhost:8181"
+	defAuthTimeout = "1s"
+
+	envLogLevel      = "MF_SMTP_NOTIFY_LOG_LEVEL"
+	envDBHost        = "MF_SMTP_NOTIFY_DB_HOST"
+	envDBPort        = "MF_SMTP_NOTIFY_DB_PORT"
+	envDBUser        = "MF_SMTP_NOTIFY_DB_USER"
+	envDBPass        = "MF_SMTP_NOTIFY_DB_PASS"
+	envDB            = "MF_SMTP_NOTIFY_DB"
+	envConfigPath    = "MF_SMTP_NOTIFY_WRITER_CONFIG_PATH"
+	envDBSSLMode     = "MF_SMTP_NOTIFY_DB_SSL_MODE"
+	envDBSSLCert     = "MF_SMTP_NOTIFY_DB_SSL_CERT"
+	envDBSSLKey      = "MF_SMTP_NOTIFY_DB_SSL_KEY"
+	envDBSSLRootCert = "MF_SMTP_NOTIFY_DB_SSL_ROOT_CERT"
+	envHTTPPort      = "MF_SMTP_NOTIFY_HTTP_PORT"
+	envServerCert    = "MF_SMTP_NOTIFY_SERVER_CERT"
+	envServerKey     = "MF_SMTP_NOTIFY_SERVER_KEY"
+	envJaegerURL     = "MF_JAEGER_URL"
+	envNatsURL       = "MF_NATS_URL"
+
+	envEmailDriver      = "MF_EMAIL_DRIVER"
+	envEmailHost        = "MF_EMAIL_HOST"
+	envEmailPort        = "MF_EMAIL_PORT"
+	envEmailUsername    = "MF_EMAIL_USERNAME"
+	envEmailPassword    = "MF_EMAIL_PASSWORD"
+	envEmailFromAddress = "MF_EMAIL_FROM_ADDRESS"
+	envEmailFromName    = "MF_EMAIL_FROM_NAME"
+	envEmailTemplate    = "MF_EMAIL_TEMPLATE"
+
+	envAuthTLS     = "MF_AUTH_CLIENT_TLS"
+	envAuthCACerts = "MF_AUTH_CA_CERTS"
+	envAuthURL     = "MF_AUTH_GRPC_URL"
+	envAuthTimeout = "MF_AUTH_GRPC_TIMEOUT"
+)
+
+type config struct {
+	natsURL     string
+	configPath  string
+	logLevel    string
+	dbConfig    postgres.Config
+	emailConf   email.Config
+	httpPort    string
+	serverCert  string
+	serverKey   string
+	jaegerURL   string
+	authTLS     bool
+	authCACerts string
+	authURL     string
+	authTimeout time.Duration
+}
+
+func main() {
+	cfg := loadConfig()
+
+	logger, err := logger.New(os.Stdout, cfg.logLevel)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	db := connectToDB(cfg.dbConfig, logger)
+	defer db.Close()
+
+	pubSub, err := nats.NewPubSub(cfg.natsURL, "", logger)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Failed to connect to NATS: %s", err))
+		os.Exit(1)
+	}
+	defer pubSub.Close()
+
+	authTracer, closer := initJaeger("auth", cfg.jaegerURL, logger)
+	defer closer.Close()
+
+	auth, close := connectToAuth(cfg, authTracer, logger)
+	if close != nil {
+		defer close()
+	}
+
+	tracer, closer := initJaeger("notify-smtp", cfg.jaegerURL, logger)
```

Comment:
here https://github.com/mainflux/mainflux/pull/1324/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R8 it is `smtp-notify`